File: app/src/main/java/com/imdmp/paperless/MainActivity.java
package com.imdmp.paperless;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.wifi.WifiManager;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;


import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.imdmp.paperless.model.Answer;
import com.imdmp.paperless.model.Event;
import com.imdmp.paperless.model.Questions;
import com.imdmp.paperless.model.SurveyTaker;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.StringWriter;
import java.io.Writer;

import static android.R.attr.button;
import static android.R.attr.data;


public class MainActivity extends AppCompatActivity {
    private static final int DEFAULT_PORT = 8080;
    private static final String TAG = "LOCAL_SERVER";


    // INSTANCE OF ANDROID WEB SERVER  = WebServer
    private BroadcastReceiver broadcastReceiverNetworkState;
    private static boolean isStarted = false;
    private WebServer adr;


    //VIEW
    Button button_startserver, button_stopserver, button_select,button_checkdb,button_test;
    TextView tv_message;
    RecyclerView rv_task;
    String textfile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {


        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //CONNECT VARIABLES
        button_startserver = (Button) findViewById(R.id.button_startserver);
        button_stopserver = (Button) findViewById(R.id.button_stopserver);
        tv_message = (TextView) findViewById(R.id.tv_message);
        rv_task = (RecyclerView) findViewById(R.id.rv_task);
        button_select = (Button) findViewById(R.id.button_select);
        button_checkdb = (Button) findViewById(R.id.button_dbcheck);
        button_test = (Button) findViewById(R.id.button_test);
        tv_message.setVisibility(View.INVISIBLE);

        button_test.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                testDatabase();
                Log.i(TAG, "testing database...");
            }
        });
        button_startserver.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //instantiate new web server
                try {
                    adr = new WebServer(DEFAULT_PORT,getBaseContext(),1);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                DatabaseHelper databaseHelper = new DatabaseHelper(getBaseContext());
                InputStream is;
                try {
                    is = getResources().getAssets().open("sample_form.html");
                    textfile = convertStreamToString(is);
                } catch (IOException e) {
                    e.printStackTrace();
                }


//                adr.setHTMLFile(textfile);
//                adr.setContext(getBaseContext());
//                adr.setDbHelper(databaseHelper);


            //  testDatabase();

                try {
                    //start
                    if (!isStarted) {
                        adr.start();
                        WifiManager wm = (WifiManager) getApplicationContext().getSystemService(WIFI_SERVICE);
                        String ip = android.text.format.Formatter.formatIpAddress(wm.getConnectionInfo().getIpAddress());
                        Log.i(TAG, "Server running at " + ip + ":8080");

                        tv_message.setText("\"Server running at \" " + ip + "\":8080\" \n Connected phones ip address: 192.168.42.1:8080");
                        tv_message.setVisibility(View.VISIBLE);
                        isStarted = true;
                    }
                } catch (IOException ioe) {
                    Log.w(TAG, "The server could not start.");
                }
            }
        });


        button_checkdb.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent dbmanager = new Intent(getBaseContext(),AndroidDatabaseManager.class);
                startActivity(dbmanager);
            }
        } );


        button_stopserver.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    //stop
                    if (isStarted) {
                        adr.stop();
                        Log.i(TAG, "Server STOPPED");
                        tv_message.setText("Server stopped.");
                        isStarted = false;
                    }
                } catch (Exception ioe) {
                    Log.w(TAG, "The server could not be stopped");
                }
            }
        });
    }

    @Override
    protected void onResume() {
        super.onResume();


    }

    // DON'T FORGET to stop the server just in case
    @Override
    public void onDestroy() {
        super.onDestroy();
        if (adr != null) {
            adr.stop();
            System.out.println("stopping server.");
        }
    }
    public void testDatabase(){
        DatabaseHelper databaseHelper = new DatabaseHelper(getBaseContext());
        databaseHelper.test();
        Event e = new Event();
        e.setFormName("test");
        databaseHelper.createEvent(e);

        int eventID = databaseHelper.getEventID(e.getFormName());
        Log.i(TAG, "eventID taken: "+eventID);
        SurveyTaker st = new SurveyTaker();
        st.setDateAnswered("September");
        st.setIdNumber("11303247");
        st.setRespondentEmail("pags@gmail.com");
        st.setRespondentName("Doms");
       // databaseHelper.createSurveyTaker(st,eventID);

        Questions q  = new Questions();
        q.setQuestion("Rate your experience so far: ");
        databaseHelper.createSurveyTaker(st,eventID);
        int surveyTakerID=  databaseHelper.getSurveyTakerID(Integer.parseInt(st.getIdNumber()));


        int questionID =  1;


        Answer a = new Answer();
        a.setAnswer("4");
        a.setQualitative(Boolean.FALSE);



        databaseHelper.addQuestion(q,eventID);
        databaseHelper.addAnswer(a,questionID);
        //q.getId();
    }

    public boolean isConnectedInWifi() {
        WifiManager wifiManager = (WifiManager) getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        NetworkInfo networkInfo = ((ConnectivityManager)  getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE)).getActiveNetworkInfo();
        if (networkInfo != null && networkInfo.isAvailable() && networkInfo.isConnected()
                && wifiManager.isWifiEnabled() && networkInfo.getTypeName().equals("WIFI")) {
            return true;
        }
        return false;
    }

    public static String convertStreamToString(InputStream is)
            throws IOException {
        Writer writer = new StringWriter();

        char[] buffer = new char[2048];
        try {
            Reader reader = new BufferedReader(new InputStreamReader(is,
                    "UTF-8"));
            int n;
            while ((n = reader.read(buffer)) != -1) {
                writer.write(buffer, 0, n);
            }
        } finally {
            is.close();
        }
        String text = writer.toString();
        return text;
    }
}

File: app/src/main/java/com/imdmp/paperless/features/MainActivity.kt
package com.imdmp.paperless.features

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.net.wifi.WifiManager
import android.os.Bundle
import android.text.format.Formatter
import android.util.Log
import android.view.View
import com.imdmp.paperless.AndroidDatabaseManager
import com.imdmp.paperless.DatabaseHelper
import com.imdmp.paperless.WebServer
import com.imdmp.paperless.databinding.ActivityMainBinding
import com.imdmp.paperless.model.Answer
import com.imdmp.paperless.model.Event
import com.imdmp.paperless.model.Questions
import com.imdmp.paperless.model.SurveyTaker
import timber.log.Timber
import java.io.*
import java.lang.Boolean
import java.lang.Exception

const val DEFAULT_PORT = 8080
class MainActivity : Activity() {

    lateinit var broadcastReceiverNetworkState :BroadcastReceiver
    var isStarted = false
    lateinit var adr :WebServer

    lateinit var binding :ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.buttonTest.setOnClickListener {
            testDatabase()
        }

        binding.buttonStartserver.setOnClickListener {
            try{
                adr = WebServer(DEFAULT_PORT,this,1)
            }catch (e:IOException){
                e.printStackTrace()
            }

            val databaseHelper  = DatabaseHelper(this)
            val inputStream  = resources.assets.open("sample_form.html")
            val file = convertStreamToString(inputStream)

//                adr.setHTMLFile(textfile);
//                adr.setContext(getBaseContext());
//                adr.setDbHelper(databaseHelper);


            //  testDatabase();

            try{
                if(isStarted.not()){
                    adr.start()
                    val wm = applicationContext.getSystemService(WIFI_SERVICE) as WifiManager
                    val ip = Formatter.formatIpAddress(wm.connectionInfo.ipAddress)
                    Timber.i(
                        "Server running at $ip:8080"
                    )

                    binding.tvMessage.setText("\"Server running at \" $ip\":8080\" \n Connected phones ip address: 192.168.42.1:8080")
                    binding.tvMessage.visibility = View.VISIBLE
                    isStarted = true

                }
            }catch (e:IOException){
                Timber.e( "The server could not start.")

            }
        }

        binding.buttonDbcheck.setOnClickListener {
            val dbmanager = Intent(this, AndroidDatabaseManager::class.java)
            startActivity(dbmanager)
        }

        binding.buttonStopserver.setOnClickListener {
            try {
                //stop
                if (isStarted) {
                    adr.stop()
                    Timber.i( "Server STOPPED")
                    binding.tvMessage.setText("Server stopped.")
                    isStarted = false
                }
            } catch (ioe: Exception) {
                Timber.e( "The server could not be stopped")
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()

        if(adr !=null){
            adr.stop()
            Timber.d("Stopping server.")
        }
    }

    fun isConnectedInWifi(): kotlin.Boolean {
        val wifiManager = applicationContext.getSystemService(WIFI_SERVICE) as WifiManager
        val networkInfo =
            (applicationContext.getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager).activeNetworkInfo
        return (networkInfo != null && networkInfo.isAvailable && networkInfo.isConnected
                && wifiManager.isWifiEnabled && networkInfo.typeName == "WIFI")
    }

    @Throws(IOException::class)
    fun convertStreamToString(`is`: InputStream): String? {
        val writer: Writer = StringWriter()
        val buffer = CharArray(2048)
        try {
            val reader: Reader = BufferedReader(
                InputStreamReader(
                    `is`,
                    "UTF-8"
                )
            )
            var n: Int
            while (reader.read(buffer).also { n = it } != -1) {
                writer.write(buffer, 0, n)
            }
        } finally {
            `is`.close()
        }
        return writer.toString()
    }

    private fun testDatabase() {
        val databaseHelper = DatabaseHelper(baseContext)
        databaseHelper.test()
        val e = Event()
        e.formName = "test"
        databaseHelper.createEvent(e)
        val eventID = databaseHelper.getEventID(e.formName)
        Timber.i("eventID taken: $eventID")
        val st = SurveyTaker()
        st.dateAnswered = "September"
        st.idNumber = "11303247"
        st.respondentEmail = "pags@gmail.com"
        st.respondentName = "Doms"
        // databaseHelper.createSurveyTaker(st,eventID);

        // databaseHelper.createSurveyTaker(st,eventID);
        val q = Questions()
        q.question = "Rate your experience so far: "
        databaseHelper.createSurveyTaker(st, eventID)
        val surveyTakerID = databaseHelper.getSurveyTakerID(st.idNumber.toInt())


        val questionID = 1
        val a = Answer()
        a.answer = "4"
        a.qualitative = Boolean.FALSE



        databaseHelper.addQuestion(q, eventID)
        databaseHelper.addAnswer(a, questionID)
        //q.getId();

    }
}